### ISSUE TITLE
Request for Detox Compatibility with React Native Version 0.71

### ISSUE THREAD

**Issue by salmaAlouane** [wix/Detox#4056]
### Description

Dear Detox Team,

I hope this message finds you well. I am a user of Detox and greatly appreciate the work you have put into this tool. However, I have recently encountered an issue when running Detox with React Native version 0.71. According to the official Detox documentation, the latest version (20.9.0) is only compatible up to React Native version 0.66.

As our project is using React Native version 0.71, I would like to inquire if there are plans to update Detox in the upcoming months to support newer versions of React Native. We would like to continue using Detox for our tests, but the current incompatibility with our React Native version is causing difficulties.

Thank you sincerely for your attention and the outstanding work you do with Detox. Your response would be greatly appreciated.

Best regards,

### Your environment

Detox version: ^20.9.0
React Native version: ^0.71.7
Node version: v18.12.1
Device model: DUB-LX1
OS:  Android
Test-runner (select one): jest

**Comment by asafkorem** [comment]
Hey @salmaAlouane, thanks for the report.
Detox is currently compatible with RN v0.70 ([see here](https://github.com/wix/Detox#react-native)), and React Native v0.71 compliance is on our plans for the near future (check https://github.com/wix/Detox/issues/3884).

Did you tried to run Detox with your project on RN v0.71 and failed? or you just assumed it won't work based on the docs?

**Comment by salmaAlouane** [comment]
Hi @asafkorem 

Thank you for your response. I did try running Detox on our project with React Native v0.71, but unfortunately, I encountered an issue. When I ran the "detox test" command to execute a simple test I had created, the application automatically uninstalled from my phone, and the test failed.

I have checked the configuration and followed the recommended steps in the Detox documentation, but the issue persists. I also verified that my development environment is properly set up.

Thank you again for your attention and ongoing support. Your response would be greatly appreciated.
